### DISCUSSION TITLE
Add «read» hooks

### DISCUSSION THREAD

**Discussion by xesjkeee** [directus/directus#4847] [Feature Requests]
For example, all my users can create an article in Admin panel and then read all articles (created by other users) in Frontend App, but in Admin App I want to display only the articles created by current user

And now I cant set «read» permissions only for Admin App, it would be spread and for REST API

It would be very useful for same cases, when I want to filter items only in Admin App, but not in Frontend App (which working on the REST API)

**Comment by xesjkeee** [comment]
refs https://github.com/directus/directus/discussions/4812
refs https://github.com/directus/directus/discussions/3736

**Comment by rijkvanzanten** [comment]
How do _read_ hooks enable you to do this?

> For example, all my users can create an article in Admin panel and then read all articles (created by other users) in Frontend App, but in Admin App I want to display only the articles created by current user

Wouldn't you be able to do this under filters in the right sidebar? 🤔 

I'm a little worried about the performance overhead of running a hook on every read

**Comment by benhaynes** [comment]
Closing as a duplicate of: https://github.com/directus/directus/discussions/3736

**Comment by benhaynes** [comment]
This feature has been approved, and will be tracked here: https://github.com/directus/directus/issues/10164
